biblib/views/document_view.py

```diff
@@ -44,20 +45,174 @@ def add_document_to_library(cls, library_id, document_data):
+                valid_bibcodes = []
+                return output_dict
+            else:
+                #If SOLR query succeeds generate list of valid bibcodes from response
+                valid_bibcodes = [doc.get('bibcode') for doc in solr_resp.get('docs', {})]
+                current_app.logger.info("Found the following valid bibcodes: {}".format(valid_bibcodes))
+            
+            if valid_bibcodes:
+                #Add all valid bibcodes to library
+                library.add_bibcodes(valid_bibcodes)
+
+                session.add(library)
+                session.commit()
+
+                current_app.logger.info('Added: {0} is now {1}'.format(
+                    valid_bibcodes,
+                    library.bibcode)
+                )
+            
+            end_length = len(library.bibcode)
 
-            current_app.logger.info('Added: {0} is now {1}'.format(
-                document_data['bibcode'],
-                library.bibcode)
-            )
+            #Generate a list of invalid bibcodes
+            invalid_bibcodes = list(set(document_data['bibcode']) - set(valid_bibcodes))
+            
+            #Generate output that contains the number added and the number of invalid bibcodes.
+            output_dict = {"number_added": end_length - start_length}
+            if invalid_bibcodes: output_dict['invalid_bibcodes'] = invalid_bibcodes
+            
+            return  output_dict
+    
+    @staticmethod
+    def _standard_ADS_bibcode_query(input_bibcodes,
+            start=0,
+            rows=20,
+            sort='date desc',
+            fl='bibcode'):
+        """
+        Validates identifiers by collecting all bibcodes returned from a standard query.
+        """
+        bibcode_query ="identifier:("+" OR ".join(input_bibcodes)+")"
+        if fl == '':
+            fl = 'bibcode'
+        else:
+            fl_split = fl.split(',')
+            for required_fl in ['bibcode']:
+                if required_fl not in fl_split:
+                    fl = '{},{}'.format(fl, required_fl)
+
+        params = {
+            'q': bibcode_query,
+            'wt': 'json',
+            'fl': fl,
+            'rows': rows,
+            'start': start,
+            'sort': sort
+        }
+
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': current_app.config.get('SERVICE_TOKEN', request.headers.get('X-Forwarded-Authorization', request.headers.get('Authorization', '')))
+        }
+        current_app.logger.info('Querying Search microservice: {0}'
+                                .format(params))
+        solr_resp = client().get(
+            url=current_app.config['BIBLIB_SOLR_SEARCH_URL'],
+            params=params,
+            headers=headers
+        )
+        return solr_resp.json(), solr_resp.status_code 
 
-            end_length = len(library.bibcode)
+    @staticmethod
+    def _solr_big_query_bibcodes(
```

Comment:
Did decide to move to base view. Had to make some minor edits to methods and names because the document_view ones were returning slightly different information than the library_view one, and the two bigquery methods were named differently, but that is all sorted.